### DISCUSSION TITLE
How to trigger reload on nginx-proxy from other container?

### DISCUSSION THREAD

**Discussion by onnimonni** [nginx-proxy/nginx-proxy#1965] [Q&A] [answered]
Hey!

I made https://github.com/onnimonni/signaler for generating ssl certificates automatically for docker containers which I use with this nginx-proxy container. I'm only using this in local development.

My problem is that the ssl generator container is not fast enough to create the ssl certificate for nginx-proxy and I need to restart the containers to retrigger nginx-proxy to read my ssl certificates again. Is there a way to reload nginx-proxy configs straight from my ssl generator container or set it up in way which watches changes in `/etc/nginx/certs`?

My whole setup can be seen in this docker-compose.yml file: https://github.com/devgeniem/gdev/blob/master/docker/services.yml

**Comment by wader** [comment]
Have a look at how docker-letsencrypt-nginx-proxy-companion does it https://github.com/JrCs/docker-letsencrypt-nginx-proxy-companion/blob/efc7e4c2d98e7f0422622ebb102d216d80dbbc66/app/functions.lib#L72

**Comment by nigelsim** [comment]
I've encountered the same issue, and I'm just making some notes about what I've found here so that either myself, or someone else can follow up on.

The best outcome would to be able to send a HUP or similar to the container and have it reload the configs. Having to know the internal setup of the container providing the proxying is fragile. 

The `docker-gen` and `nginx` containers each seem to support this behaviour out of the box, but the `nginx-proxy` container, which uses [Forego](https://github.com/ddollar/forego) [does not](https://github.com/ddollar/forego/issues/114).

Additionally, in my particular configuration I'm updating the virtual host specific files which are `include`d in the `/etc/nginx/conf.d/default.conf`, meaning that `docker-gen` does not detect the change, so does not signal `nginx`.

**Comment by Dr-Shadow** [comment, marked as answer]
For anyone having issues with reload, you can signal the process you need with these commands :

```
docker exec -it nginx-proxy sh -c 'kill -SIGHUP $(pidof docker-gen)'
docker exec -it nginx-proxy sh -c 'kill -SIGHUP $(cat /run/nginx.pid)'
```

Of course, you will probably need to escape the $ or ' depending on your use case (in scripts, etc...)
